cli/mqtt.py

```diff
@@ -13,10 +13,13 @@
 
 def mqtt_open(env):
     with env.config.open() as cfg:
-        printer = cfg.printers[0]
+        if env.printer >= len(cfg.printers):
```

Comment:
It's great that we're doing error checking, but we shouldn't be duplicating code (between here and the pppp couterpart).

There's an easy fix! The `env` is of type `Environment` (see the top of this file), and it's made for exactly this kind of task.

Look at the comment for that class, to see how to extend it.

After it's fixed, we can simply refer to the printer through `env` :)

Reply:
If we view the efficiency loss of having to import the config multiple times if/when necessary to be negligible, I can go ahead and implement the requested change. I also like the idea of potentially caching the config and referencing it from cache when needed.